examples/jsm/nodes/core/NodeBuilder.js

```diff
@@ -1141,6 +1170,15 @@ class NodeBuilder {
 
 		}
 
+		if ( fromTypeLength === 1 && toTypeLength > 1 && fromType[ 0 ] !== toType[ 0 ] ) { // fromType is float-like
```

Comment:
What are the cases that don't work without this clause?

Reply:
Is it done exclusively to make something like `vec4( x )` output `vec4( x )` instead of `vec4( vec3( x ), 1.0 )`? If so, I think this clause isn't needed, only the `&& fromTypeLength > 1` above is sufficient.